oauth2_http/javatests/com/google/auth/oauth2/Slf4jUtilsTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.auth.oauth2;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.auth.oauth2.Slf4jUtils.LoggerFactoryProvider;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.ILoggerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.helpers.NOPLogger;
+
+public class Slf4jUtilsTest {
+  private static final Logger LOGGER = LoggerFactory.getLogger(Slf4jUtilsTest.class);
+
+  private TestEnvironmentProvider testEnvironmentProvider;
+
+  @Before
+  public void setup() {
+    testEnvironmentProvider = new TestEnvironmentProvider();
+  }
+  // This test mimics GOOGLE_SDK_JAVA_LOGGING != true
+  @Test
+  public void testGetLogger_loggingDisabled_shouldGetNOPLogger() {
+    testEnvironmentProvider.setEnv(LoggingUtils.GOOGLE_SDK_JAVA_LOGGING, "false");
+    LoggingUtils.setEnvironmentProvider(testEnvironmentProvider);
+    Logger logger = Slf4jUtils.getLogger(Slf4jUtilsTest.class);
+
+    assertEquals(NOPLogger.class, logger.getClass());
+    assertFalse(logger.isInfoEnabled());
+    assertFalse(logger.isDebugEnabled());
+  }
+
+  // This test require binding (e.g. logback) be present
+  @Test
+  public void testGetLogger_loggingEnabled_slf4jBindingPresent() {
```

Comment:
From the test result, I can see the binding is not present.

Could you change the test name to reflect that?

Reply:
I am not sure I get what you mean?

This test is only included when `slf4j2x-test` is active (logback present), so I expect`assertNotEquals(NOPLogger.class, logger.getClass());`

Reply:
Ah, my bad, I thought it's `assertEquals`.